[CMake] llm_runner use executorch_core

kirklandsign opened the pull request:
### Summary
We only need executorch_core. We don't need prim ops for runner library. runner.exe can add prim_ops separately

Change:
```
set(_extension_llm_runner__srcs
-    extension/tensor/tensor_ptr.cpp
-    extension/tensor/tensor_ptr_maker.cpp
)```

### Test plan
CI

cc @larryliu0820 @jathu

kirklandsign committed:
[CMake] Update extension/ deps executorch->executorch_core

kirklandsign committed:
test more

kirklandsign committed:
Add deps

kirklandsign committed:
Don't address flat_tensor yet

kirklandsign committed:
Merge remote-tracking branch 'origin/main' into use-executorch-core

kirklandsign committed:
Try again

lucylq commented on tools/cmake/cmake_deps.toml:
```diff
@@ -197,8 +196,7 @@ filters = [
   ".cpp$",
 ]
 deps = [
-  "extension_flat_tensor_schema",
```
was this unecessary?

kirklandsign replied:
Let me add it back. Probably shouldn't remove.

kirklandsign committed:
missing

kirklandsign committed:
Update CMakeLists.txt

kirklandsign committed:
Merge remote-tracking branch 'origin/main' into use-executorch-core

kirklandsign committed:
module update

kirklandsign committed:
Fix

kirklandsign committed:
Merge remote-tracking branch 'origin/main' into use-executorch-core

kirklandsign committed:
flat tensor doesn't dep on extension_data_loader

kirklandsign committed:
linter

kirklandsign committed:
[CMake] Avoid extension_module have dupe flat_tensor cpp

Currently, _extension_module__srcs has srcs from flat_tensor. This is
bad.

```
set(_extension_module__srcs
    extension/flat_tensor/flat_tensor_data_map.cpp
    extension/flat_tensor/serialize/flat_tensor_header.cpp
    extension/module/module.cpp
)
```

We need to update deps. We need to export extension_flat_tensor to
install directory.

kirklandsign committed:
propogate deps

kirklandsign committed:
Kernels should depend on executorch_core

kirklandsign committed:
Add executorch to runner.exe

kirklandsign committed:
Merge branch 'kernel-use-core' into module-avoid-dupe-cpp

kirklandsign committed:
Update runner to include extension_flat_tensor

kirklandsign committed:
Merge branch 'module-avoid-dupe-cpp' into use-executorch-core

kirklandsign committed:
Merge remote-tracking branch 'origin/main' into module-avoid-dupe-cpp

kirklandsign committed:
add extension_flat_tensor

kirklandsign committed:
Merge branch 'module-avoid-dupe-cpp' into use-executorch-core

kirklandsign committed:
untouch

kirklandsign committed:
Merge remote-tracking branch 'origin/main' into use-executorch-core
